net/interfaces/interfaces.go

```diff
@@ -117,17 +121,43 @@ func LocalAddresses() (regular, loopback []netaddr.IP, err error) {
 				if tsaddr.IsTailscaleIP(ip) {
 					continue
 				}
-				if ip.IsLinkLocalUnicast() {
-					continue
-				}
 				if ip.IsLoopback() || ifcIsLoopback {
 					loopback = append(loopback, ip)
+				} else if ip.Is4() && ip.IsLinkLocalUnicast() &&
+					hostinfo.GetEnvType() == hostinfo.AWSLambda {
```

Comment:
let's put this down on line 152 instead. 

Also, right now LinkLocalUnicast IPv6 can make their way down to regular6 it looks like. 

So instead I'd make this block only about LinkLocalUnicast:

```go
    } else if ip.IsLinkLocalUnicast() {
          if ip.Is4() {
             linklocal4 = append(linklocal4, ip)
          } 
    } ...
```

Then you're just populating info about addresses found and all the policy of what to return as a function of the environment can be down below in one place, around line 152.

Reply:
Done.